homeassistant/components/hue.py

```diff
@@ -0,0 +1,237 @@
+    try:
+        bridge = phue.Bridge(
+            host,
+            config_file_path=hass.config.path(PHUE_CONFIG_FILE))
+        bridge_id = bridge.get_api().get('config').get(ATTR_HUE_BRIDGEID)
+    except (
+            ConnectionRefusedError,
+            OSError):
+        _LOGGER.error("Error connecting to the Hue bridge at %s", host)
+        return
+    except phue.PhueRegistrationException:
+        _LOGGER.warning("Connected to Hue at %s but not registered.", host)
+        request_configuration(hass, host)
+        return
+
+    # If we came here and configuring this host, mark as done
+    if host in _CONFIGURING:
+        request_id = _CONFIGURING.pop(host)
+        configurator = get_component('configurator')
+        configurator.request_done(request_id)
+
+    # Add instanciated bridge to our own control global
+    if HUEBRIDGE.get(bridge_id):
+        _LOGGER.info('Bridge %s already initialized, skipping', bridge_id)
+        return
+    HUEBRIDGE[bridge_id] = HueBridge(hass, bridge)
+
+
+def request_configuration(hass, host):
+    """ Request configuration steps from the user. """
+    configurator = get_component('configurator')
+
+    # We got an error if this method is called while we are configuring
+    if host in _CONFIGURING:
+        configurator.notify_errors(
+            _CONFIGURING[host], "Failed to register, please try again.")
+        return
+
+    # pylint: disable=unused-argument
+    def hue_configuration_callback(data):
+        """ Actions to do when our configuration callback is called. """
+        setup_huebridge(hass, host)
+
+    _CONFIGURING[host] = configurator.request_config(
+        hass, "Philips Hue", hue_configuration_callback,
+        description=("Press the button on the bridge to register Philips Hue "
+                     "with Home Assistant."),
+        description_image="/static/images/config_philips_hue.jpg",
+        submit_caption="I have pressed the button"
+    )
+
+
+class HueBridge(object):
+    """
+    This manages a Hue bridge
+    Gets the latest data and update the states
+    """
+
+    def __init__(self, hass, bridge):
+        # This will be used to control devices from their platform modules
+        self.hass = hass
+        self._bridge = bridge
+
+        config = self._fetch_apidata().get('config')
+        self.bridge_id = config.get(ATTR_HUE_BRIDGEID)
+        self.bridge_ip = config.get(ATTR_HUE_BRIDGEIP)
+
+        # Support limitations in DiY zigbee bridge 'deconz'
+        # FIXME / This is based on the name, a user configureable parameter?
+        if config.get('name') == 'RaspBee-GW':
+            self.bridge_type = 'deconz'
+        else:
+            self.bridge_type = 'hue'
+
+        self.states = {}
+        self.update()
+
+        # FIXME / debuglog
+        _LOGGER.warning('New bridge: %s/%s', self.bridge_ip, self.bridge_id)
+
+    def set_light(self, light_id, command):
+        self._bridge.set_light(int(light_id), command)
```

Comment:
Need to cast to int(). Bridge states use number strings, bridge get_light expects int.
Weird, this wasn't needed in the old code.